vortex-array/src/view.rs

```diff
@@ -136,8 +141,198 @@ impl<'v> ArrayView<'v> {
+            let is_constant = array.stats()?.is_constant();
+            is_constant
+                .and_then(|v| v.type__as_bool())
+                .map(|v| v.value().into())
+        }
+        Stat::IsSorted => array
+            .stats()?
+            .is_sorted()
+            .and_then(|v| v.type__as_bool())
+            .map(|v| v.value().into()),
+        Stat::IsStrictSorted => array
+            .stats()?
+            .is_strict_sorted()
+            .and_then(|v| v.type__as_bool())
+            .map(|v| v.value().into()),
+        Stat::Max => {
+            let max = array.stats()?.max();
+            max.and_then(|v| v.type__as_primitive())
+                .and_then(primitive_to_scalar)
+        }
+        Stat::Min => {
+            let min = array.stats()?.min();
+            min.and_then(|v| v.type__as_primitive())
+                .and_then(primitive_to_scalar)
+        }
+        Stat::RunCount => {
+            let rc = array.stats()?.run_count();
+            rc.and_then(|v| v.type__as_primitive())
+                .and_then(primitive_to_scalar)
+        }
+        Stat::TrueCount => {
+            let tc = array.stats()?.true_count();
+            tc.and_then(|v| v.type__as_primitive())
+                .and_then(primitive_to_scalar)
+        }
+        Stat::NullCount => {
+            let nc = array.stats()?.null_count();
+            nc.and_then(|v| v.type__as_primitive())
+                .and_then(primitive_to_scalar)
+        }
+        Stat::BitWidthFreq => array
+            .stats()?
+            .bit_width_freq()
+            .map(|v| {
+                v.iter()
+                    .flat_map(|v| {
+                        primitive_to_scalar(
+                            v.type__as_primitive()
+                                .expect("Should only ever produce primitives"),
+                        )
+                    })
+                    .collect_vec()
+            })
+            .map(|v| {
+                List(ListScalar::new(
+                    DType::Primitive(vortex_dtype::PType::U64, Nullability::NonNullable),
+                    Some(v),
+                ))
+            }),
+        Stat::TrailingZeroFreq => array
+            .stats()?
+            .trailing_zero_freq()
+            .map(|v| {
+                v.iter()
+                    .flat_map(|v| {
+                        primitive_to_scalar(
+                            v.type__as_primitive()
+                                .expect("Should only ever produce primitives"),
+                        )
+                    })
+                    .collect_vec()
+            })
+            .map(|v| {
+                List(ListScalar::new(
+                    DType::Primitive(vortex_dtype::PType::U64, Nullability::NonNullable),
+                    Some(v),
+                ))
+            }),
+    }
+}
+
+// TODO(@jcasale): move this to serde and make serde crate public?
+fn primitive_to_scalar(v: Primitive) -> Option<Scalar> {
```

Comment:
similar to commented-out code [here](https://github.com/spiraldb/vortex/blob/5143c3f1cd7944c7ffe8e6f7a3879e44514c5067/vortex-scalar/src/serde.rs#L206), but does not rely on a reader. Maybe we can delete that code?

Reply:
This sort of does already exist. You should be able to do `Scalar::read_flatbuffer(&v)`